Forgot password feature + strong password enforcement

claynaut opened the pull request:
<!-- Please follow the provided template -->
### 1. Please briefly explain what it is you coded
Implemented the forgot password feature and enforced strong passwords whenever making a password is involved (i.e. user registration and password reset).

### 2. Please briefly explain your approach (new components added? modified? removed?)
Added a forgot password request form. Modified graphql types to incorporate a new method/resolver to reset a user's password via email.

Added a reset password form and modified the existing reset password page to incorporate the new form with a similar UI as the forgot password page. Added new graphql types to incorporate the reset password feature with token verification.

Enforced strong password validation using regex.

### 3. Any packages added/updated/removed
N/A
### (Optional) 4. Any feature recommendations/insights/questions/comments/etc.?

Commit:
Revert "feat(frontend): implement dashboard"

This reverts commit c88482c77584ba8ba56f7ae119ac499c8e1815ff.

Commit:
Revert "feat(frontend): implement dashboard"

This reverts commit c88482c77584ba8ba56f7ae119ac499c8e1815ff.

Commit:
Revert "feat(frontend): implement dashboard"

This reverts commit c88482c77584ba8ba56f7ae119ac499c8e1815ff.

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
feat(backend): enforce strong password validation for user registration

Commit:
fix(frontend): update password criteria for user registration

Commit:
fix(frontend): update password criteria for user registration

Commit:
fix(frontend): update password criteria for user registration

Commit:
feat(project): implement forgot password feature

Commit:
feat(project): implement forgot password feature

Commit:
feat(project): implement forgot password feature

Commit:
feat(project): implement forgot password feature

DevelopmentByDavid committed:
feat(server): added appropriate graphql types for question queue redesign

DevelopmentByDavid committed:
feat(server): first pass at implementation of QQRedesign

DevelopmentByDavid committed:
feat(server): added support for question queue pagination

DevelopmentByDavid committed:
feat(client): question carousel supports new subscription design

DevelopmentByDavid committed:
feat(client): add support for updated subscription design to question list

DevelopmentByDavid committed:
feat(client): add support for updated subscription design to question queue

kadey001 commented:
Looks good, but when clicking on "forgot password" it should just prompt them for an email, then there should be another page for reset-password that takes in a token (ie. prytaneum.io/reset-password?token=1234). The reset-password page would use that token to validate the password reset is valid, letting them put in the new password + confirmation and update it. If the token is invalid it should display a snack saying the token is invalid/expired.

kadey001 reviewed (changes requested):
Looks good, but when clicking on "forgot password" it should just prompt them for an email, then there should be another page for reset-password that takes in a token (ie. prytaneum.io/reset-password?token=1234). The reset-password page would use that token to validate the password reset is valid, letting them put in the new password + confirmation and update it. If the token is invalid it should display a snack saying the token is invalid/expired.

Commit:
chore(project): resolve conflicts

Commit:
feat(project): implement proper password reset requests with token verification

Commit:
feat(project): implement proper password reset requests with token verification

Commit:
chore(project): update comments

Commit:
chore(project): update comments

kadey001 committed:
Merge remote-tracking branch 'c2c/dev' into dev

kadey001 committed:
Merge branch 'dev' into CI/CD

kadey001 committed:
Merge remote-tracking branch 'c2c/dev' into CI/CD

kadey001 committed:
ci(formatted + dev deployment): namespace more explicit now

kadey001 committed:
Merge pull request #3 from kadey001/CI/CD

CI/CD Pipeline for Dev & Production + Split out frontend and backend

kadey001 committed:
ci(formatting): fixed yml formatting

kadey001 committed:
fix(dev deployment): fixed tag var

kadey001 committed:
Merge branch 'CI/CD' into dev

kadey001 committed:
fix(updated kustomize file): removed resource that was moved

kadey001 committed:
fix(kubernetes): updated db-secret file name

kadey001 committed:
ci(ingress): updated ingress file locations & added a dev version

kadey001 committed:
ci(ingress): updated ingress file locations & added a dev version

kadey001 committed:
Merge branch 'dev' of https://github.com/kadey001/prytaneum into dev

kadey001 committed:
ci(action name update): fixed dev version name

kadey001 committed:
ci(typo): fix ingress file name

kadey001 committed:
ci(dev): updated regional static ip name for dev deployment

kadey001 committed:
ci(prod fix): temp branch change for prod fix

kadey001 committed:
ci(typo+dev): updated ingress for dev

kadey001 committed:
ci(updated dev deployment): updated client for dev deployment

kadey001 committed:
ci(dev deployment): removed quotes from DEV_GRAPHQL secret

kadey001 committed:
ci(dev): updated ingress

kadey001 committed:
ci(dev): re-added static-ip name

kadey001 committed:
ci(dev): updated GRAPHQL_URL & fixed typo in dockerifle

claynaut committed:
Merge branch 'dev' into password-features

kadey001 committed:
Merge pull request #141 from kadey001/dev

Set up Cloud Deployment pipeline

DevelopmentByDavid committed:
refactor(server): start and end cursor are more robust slight formatting fix

DevelopmentByDavid committed:
feat(project): subscriptions are fixed and mass relay update server & client

DevelopmentByDavid committed:
feat(client): moderator dnd works properly via subscriptions

DevelopmentByDavid committed:
fix(client): small logic errors

DevelopmentByDavid committed:
fix(project): introduce new edge wrapper and copy/paste error fixes

kadey001 committed:
chore(env): updated to local vars

kadey001 committed:
feat(backend): added resolvers and methods for livefeedback

kadey001 committed:
feat(frontend): added live feedback ui, fragments, and mutations

kadey001 committed:
fix(live feedback): fixed live feedback resolver and subscription

kadey001 committed:
ci(gke): updated dev ingress to use dev.prytaneum.io

kadey001 committed:
ci(gke): updated managed cert adding dev subdomain

kadey001 committed:
fix(live feedback): fixed feedback order and moved list filtering

kadey001 committed:
Revert "ci(gke): updated dev ingress to use dev.prytaneum.io"

This reverts commit 28ec45d9fa1012df10b9519dd5948cd0c0858e08.

kadey001 committed:
Merge pull request #143 from kadey001/live-feedback

Live feedback feature

kadey001 committed:
Revert "Live feedback"

kadey001 committed:
Merge pull request #144 from c2c-project/revert-143-live-feedback

Revert "Live feedback"

kadey001 committed:
Revert "Revert "Live feedback""

kadey001 committed:
fix(update moderator): fixed build problem with unimplemented feature

kadey001 committed:
Merge pull request #146 from kadey001/hotfix

Build fix

kadey001 committed:
Merge pull request #145 from c2c-project/revert-144-revert-143-live-feedback

Revert the Revert, restoring the Live Feedback merge post hotfix.

kadey001 committed:
Fixed UpdateModerator type

kadey001 committed:
Fixed updateModerator

kadey001 committed:
Merge branch 'dev' of https://github.com/c2c-project/prytaneum into dev

kadey001 committed:
env fix

fixed db env file

kadey001 committed:
fix(db url env): fixed db env issue with deployment/development

kadey001 committed:
Merge branch 'dev' into db-env-deployment

kadey001 committed:
Merge pull request #147 from kadey001/db-env-deployment

fixed the db url so now it works when developing locally as well as providing the correct url when deploying via the dockerfile env var.

kadey001 committed:
Revert "Db env deployment"

kadey001 committed:
Merge pull request #148 from c2c-project/revert-147-db-env-deployment

Revert "Db env deployment"

kadey001 committed:
fix(env): updated database url env var to string w/o \

kadey001 committed:
Merge branch 'db-env-deployment' of https://github.com/kadey001/prytaneum into db-env-deployment

kadey001 committed:
Merge branch 'dev' into db-env-deployment

kadey001 committed:
Merge pull request #149 from kadey001/db-env-deployment

Db env deployment

kadey001 committed:
fix(feedback): updated to require being logged in to view live feedback

kadey001 committed:
build(next): updated next version

kadey001 committed:
fix(env): removed old unsued env vars

kadey001 committed:
build(typescript): updated typescript version to fix next errors

kadey001 committed:
fix(env): dev env fix

kadey001 committed:
chore(footer): temporarly removed footer until it is fixed

kadey001 committed:
Merge pull request #150 from kadey001/dev

Minor Fixes

kadey001 committed:
Update .env

Trying to fix production env issue

kadey001 committed:
Update .env

Temporarily reverting env changes to fix production build's connection to the database

kadey001 committed:
Update production-deployment.yml

Deploying to production by temporarily changing workflow since merging to master could cause issues currently

kadey001 committed:
Updated production-deployment.yml

Updated tag for docker images

kadey001 committed:
Update relay.ts

Updated wsProtocal to only wss to try and fix production issue.

kadey001 committed:
Updated production-deployment.yml

Changed production deployment branch trigger back to master

kadey001 committed:
ci(deployment): trying to use pryatneum dev subdomain

kadey001 committed:
ci(dockerfile): added default env db vars and removed from env file

kadey001 committed:
ci(deployment): reverted dev deployment ingress

kadey001 committed:
ci(deployment): updaed managed cert name

kadey001 committed:
ci(certificate): added development certificate

kadey001 committed:
ci(certificate): fixed namespace for dev cert

kadey001 committed:
ci(deployment): updated database url env

kadey001 committed:
ci(env): updated db url to single secret var

kadey001 committed:
chore(packages): removed prytaneum typings

kadey001 committed:
fix(position): fixed position bug by increasing time slice used

kadey001 committed:
chore(dockerfile): cleaned up unessisary env vars

kadey001 committed:
Merge pull request #153 from kadey001/dev

Dev Deployment, Dev Environment Fix, and Bug Fix

kadey001 committed:
chore(redux): removing redux

kadey001 committed:
fix(ws): only use wss in prodution env

kadey001 committed:
fix(event sidebar): fixed sidebar tabs

kadey001 committed:
ci(deployment): updated to deploy on pull requests to dev

kadey001 committed:
ci(deployment): updated to deploy on pull requests to dev

kadey001 committed:
Merge branch 'unused-code-cleanup' of https://github.com/kadey001/prytaneum into unused-code-cleanup

kadey001 committed:
ci(deployment): removed on pull request

kadey001 committed:
Merge pull request #160 from kadey001/unused-code-cleanup

Event Sidebar Tabs Refactor & Redux Removal

kadey001 committed:
Update development-deployment.yml

Updated Grahpql url

kadey001 committed:
fix(card size bug): long words now break and wrap correctly within a question card

kadey001 committed:
chore(event sidebar): updated default tab

kadey001 committed:
fix(tabs): made tabs scrollable so all can be seen on mobile

kadey001 committed:
fix(user settings): cleaned up and fixed logged out issue on user settings page

kadey001 committed:
fix(logout): fixed user menu logout with useLogout hook

kadey001 committed:
fix(login): updated login onSuccess route

kadey001 committed:
fix(protected routes): protected routes depending on users login status

kadey001 committed:
chore(cleanup): code cleanup

kadey001 committed:
Merge branch 'dev' of https://github.com/kadey001/prytaneum into next-update

kadey001 committed:
ci(deployment): deploy on pull request target

kadey001 committed:
chore(next + lint): upgraded next version and added linting for the client

kadey001 committed:
chore(useLogout): code cleanup + relay compile fix

kadey001 committed:
fix(user context): refreshing on protected routes now works properly

kadey001 committed:
chore(eslint): updated file extension for consistancy

kadey001 committed:
chore(eslintrc): added next to extends field

kadey001 committed:
ci(linting): added linting to github actions workflow

kadey001 committed:
ci(linting): updated script

kadey001 committed:
ci(linting): updated steps

kadey001 committed:
ci(linting): testing max-warnings flag

kadey001 committed:
fix(linting): fixed some linter warnings

kadey001 committed:
chore(husky): added linting to husky pre-commit

kadey001 committed:
fix(next): downgraded next version slightly to fix yarn pnp issue

kadey001 committed:
fix(eslint): downgraded eslint to v7 to work with next version

kadey001 committed:
Merge pull request #163 from kadey001/next-update

Next + Typescript Upgrade & Client Linting

kadey001 committed:
ci(env): added mailgun env vars to deployment

kadey001 committed:
feat(logger): added winston logger

kadey001 committed:
chore(logging): updated console logs to use logger

kadey001 committed:
fix(service worker): added unregistering for old service workers

kadey001 committed:
fix(eslint): fixed eslint error with service worker unregistering

kadey001 committed:
fix(sidenav): disabled movingbg component to address mobile ui problem

kadey001 committed:
Merge branch 'dev' into dev

kadey001 committed:
Merge pull request #161 from kadey001/dev

UI Bug Fixes + Service Worker Fix

kadey001 committed:
ci(workflow): updated deployment workflow to only deploy on a pull request merge

kadey001 committed:
Merge branch 'dev' of https://github.com/c2c-project/prytaneum into dev

kadey001 committed:
Merge pull request #168 from kadey001/dev

Workflow Fix

kadey001 committed:
Merge branch 'dev' into login/logout-bugfix

kadey001 committed:
Merge pull request #162 from kadey001/login/logout-bugfix

Login/logout Bug Fixes

kadey001 committed:
ci(workflow): change the trigger

updated on pull_request to on pull_request_target

kadey001 committed:
ci(workflow): updated merge linting triggers

kadey001 committed:
chore(cleanup): removed unused commented code

kadey001 committed:
Merge pull request #169 from kadey001/dev

Workflow Test PR

kadey001 committed:
ci(workflow): removed a trigger

This extra trigger was causing the workflow to be run 2x as a duplicate.

kadey001 committed:
chore(winston): updated prod and dev loggers

updated build functions and naming conventions for loggers

kadey001 committed:
chore(server logging): updated server logging to use winston

errors and their stacktrace should now be logged when they happen.

kadey001 committed:
Merge pull request #170 from kadey001/logging

Improved Logging

kadey001 committed:
ci(linting): added back trigger

only the one trigger did not seem to work so the trigger is being re-added

kadey001 committed:
chore(formatting): updated workflows to prettier format

kadey001 committed:
Merge pull request #171 from kadey001/dev

Workflow Test

kadey001 committed:
chore(logging): added gcloud winston transport

kadey001 committed:
chore(logger): added service context

kadey001 committed:
Merge pull request #172 from kadey001/logging

GKE Logging

kadey001 committed:
ci(workflow): dev workflow refactor

kadey001 committed:
ci(workflow): removed old workflow files

kadey001 committed:
Merge pull request #173 from kadey001/workflow-refactor

Workflow refactor

kadey001 committed:
chore(eslint): added eslint packages and script to server

kadey001 committed:
fix(lint): fixed linting error

kadey001 committed:
chore(eslint): updated script to have max warnings set as 10

kadey001 committed:
ci(workflow): added server linting step

kadey001 committed:
fix(linting): fixed a number of linting warnings

kadey001 committed:
Merge pull request #174 from kadey001/server-linting

Server linting

kadey001 committed:
ci(workflow): updated workflow

trying a different condition to see if deployment will trigger properly.

kadey001 committed:
ci(workflow): updated dev workflow

still attempting to have deployment run on pr merge

kadey001 committed:
Workflow Fix (#175)

updated workflow to try and get deployment triggering on merge

kadey001 committed:
Revert "Workflow Fix (#175)"

This reverts commit f20850a244172463cd345c181df866a787b7f40f.

kadey001 committed:
Workflow fix (#177)

* ci(workflow): updated workflow

kadey001 committed:
Revert "Workflow fix (#177)"

This reverts commit 7cf72cd37aea4e29031f42ec002f8c127d096b0c.

kadey001 committed:
Merge pull request #178 from c2c-project/revert-177-workflow-fix

Revert "Workflow fix"

kadey001 committed:
Workflow fix (#179)

* ci(workflow): updated workflow

kadey001 committed:
ci(workflow): removed a trigger

Removed the pull_request trigger since it fails on PRs from a forked repo

kadey001 committed:
fix(feedback card): text now wraps properly

kadey001 committed:
Merge pull request #180 from kadey001/live-feedback-bugfix

Feedback Card Bugfix

kadey001 committed:
ci(workflow): updated workflow name

shortened workflow name

kadey001 committed:
fix(logout): fixed main logout bugs

kadey001 committed:
style(useLogout): minor style changes and removing unused imports

kadey001 committed:
refactor(org profile): removed redundant hook

parent already checks for users auth status

kadey001 committed:
Merge pull request #182 from kadey001/#181-logout-bugfixes

#181 logout bugfixes

kadey001 committed:
chore(fastify): updated fastify version + added uuid package

kadey001 committed:
build(logging): updated to use fastify logger for all logging

updated data that is output with the logs making them more informative and useful

kadey001 committed:
build(winston): removed winston logger

winston logger has been replaced by the built in fastify pino logger

kadey001 committed:
Merge branch 'c2c-project:dev' into mailgun

kadey001 committed:
Merge branch 'c2c-project:dev' into server-logging

kadey001 committed:
ci(workflow): updated dev workflow triggers

Removed the on push trigger as workflow only really needs to run on the pull_request_target events. It is redundant to also trigger them on pushes

kadey001 committed:
Merge branch 'c2c-project:dev' into server-logging

kadey001 committed:
Merge branch 'c2c-project:dev' into mailgun

kadey001 committed:
refactor(mailgun): simplified sendEmail function to make it more generic and reuseable

kadey001 committed:
refactor(email invite): cleaned up commented code and updated email invite to use a template

kadey001 committed:
Merge pull request #183 from kadey001/server-logging

Improved Server Logging

kadey001 committed:
Merge branch 'c2c-project:dev' into mailgun

kadey001 committed:
Merge pull request #184 from kadey001/mailgun

Mailgun Configuring/Refactor

kadey001 committed:
ci(workflow): removed duplicate event trigger

kadey001 committed:
fix(question list length): increased default amount of questions fetched as temp fix

kadey001 committed:
Merge pull request #188 from kadey001/dev

Question List Default Length Increase

DevelopmentByDavid committed:
refactor(project): improved moderator actions

kadey001 committed:
feat(prisma): added live feedback reply fields to db model

kadey001 committed:
feat(schema): added live feedback reply fields to schema

kadey001 committed:
feat(server): implemented resolver for live feedback reply

kadey001 committed:
feat(client): implemented live feedback reply on the client

kadey001 committed:
chore(cleanup): removed unused import and commented code

kadey001 committed:
fix(console error): fixed a console error

DevelopmentByDavid committed:
feat(project): moderator actions now work

kadey001 committed:
feat(logo): temporarily removed navigation on logo click

kadey001 committed:
fix(carousel): fixed subscriptions for the question carousel

kadey001 committed:
chore(cleanup): removed unused code and updated default queue ammount

kadey001 committed:
refactor(question list/carousel): updated question list + carousel to use updated subscriptions

kadey001 committed:
Merge pull request #142 from DevelopmentByDavid/dev-subscriptions

Fix graphql subscriptions

kadey001 committed:
fix(build error): fixed build error

kadey001 committed:
Updated workflow temporarily

kadey001 committed:
chore(cleanup): code cleanup

kadey001 committed:
Merge branch 'dev' of https://github.com/c2c-project/prytaneum into remote-dev

kadey001 committed:
Merge branch 'c2c-project:dev' into dev

kadey001 committed:
Merge pull request #200 from kadey001/dev

Minor Fixes/Changes

kadey001 committed:
fix(question carousel): uncommented subscriptions

kadey001 committed:
ci(backed-config): added backend config for longer timeout

kadey001 committed:
Merge branch 'c2c-project:dev' into dev

kadey001 committed:
Merge pull request #201 from kadey001/dev

Question Carousel Subscriptions

kadey001 committed:
Merge branch 'c2c-project:dev' into ws-fix

kadey001 committed:
Merge pull request #202 from kadey001/ws-fix

Websocket Timeout Fix

kadey001 committed:
feat(question list): increased default number of questions fetched

kadey001 committed:
ci(backend config): updated kustomize with backed config

kadey001 committed:
Merge pull request #203 from kadey001/dev

Quest List + WS Backed Config

kadey001 committed:
Merge branch 'dev' of https://github.com/c2c-project/prytaneum into live-feedback-reply

kadey001 committed:
chore(cleanup): removed unused import

kadey001 committed:
Merge pull request #196 from kadey001/live-feedback-reply

Live Feedback Reply

kadey001 committed:
fix(ws client): updated config for the ws client

kadey001 committed:
Merge pull request #211 from kadey001/dev

Attempted Websocket Fix

kadey001 committed:
ci(backend config): added backend config to ingress

kadey001 committed:
revert(subscription client): removed some options

kadey001 committed:
Merge branch 'c2c-project:dev' into dev

kadey001 committed:
Merge pull request #212 from kadey001/dev

Added Backend Config to Ingress

claynaut committed:
chore(project): clean up code from rebase

claynaut committed:
fix(frontend): fix login modal not closing when switching pages
